packages/swapper/src/swappers/thorchain/utils/getTradeRate/getTradeRate.ts

```diff
@@ -0,0 +1,78 @@
+import { adapters } from '@shapeshiftoss/caip'
+import { Asset } from '@shapeshiftoss/types'
+
+import { SwapError, SwapErrorTypes } from '../../../../api'
+import { BN, bn } from '../../../utils/bignumber'
+import { PoolResponse, ThorchainSwapperDeps } from '../../types'
+import { fromBaseUnit, toBaseUnit } from '../ethereum/makeTradeTx'
+import { thorService } from '../thorService'
+
+const THOR_PRECISION = 8
+
+type PoolData = {
+  assetBalance: BN
+  runeBalance: BN
+}
+
+const getSwapOutput = (inputAmount: BN, poolData: PoolData, toRune: boolean): BN => {
+  const x = inputAmount
+  const X = toRune ? poolData.assetBalance : poolData.runeBalance
+  const Y = toRune ? poolData.runeBalance : poolData.assetBalance
+  const numerator = x.times(X).times(Y)
+  const denominator = x.plus(X).pow(2)
+  return numerator.div(denominator)
+}
+
+const getDoubleSwapOutput = (input: BN, inputPool: PoolData, outputPool: PoolData): BN => {
+  const runeToOutput = getSwapOutput(input, inputPool, true)
+  return getSwapOutput(runeToOutput, outputPool, false)
+}
+
+// https://docs.thorchain.org/how-it-works/prices
+// TODO this does not support swaps between native "RUNE"
+// Rune swaps use a different calculation because its 1 hop between pools instead of 2
+export const getTradeRate = async (
+  sellAsset: Asset,
+  buyAsset: Asset,
+  sellAmount: string,
+  deps: ThorchainSwapperDeps
+): Promise<string> => {
+  const buyPoolId = adapters.assetIdToPoolAssetId({ assetId: buyAsset.assetId })
+  const sellPoolId = adapters.assetIdToPoolAssetId({ assetId: sellAsset.assetId })
+
+  if (!buyPoolId || !sellPoolId)
+    throw new SwapError(`[getPriceRatio]: No thorchain pool found`, {
+      code: SwapErrorTypes.RESPONSE_ERROR,
```

Comment:
Do you not have an `INVALID_POOL` or similar error rather than a generic response error?

Reply:
can add one